src/bicep/modules/privateLink.bicep

```diff
@@ -22,10 +22,10 @@ param vnetResourceGroup string = resourceGroup().name
 @description('The subscription id of the subscription the virtual network exists in')
 param vnetSubscriptionId string = subscription().subscriptionId
 
-var privateLinkConnectionName  = substring('plconn${logAnalyticsWorkspaceName }${uniqueData}', 0, 16)
-var privateLinkEndpointName = substring('pl${logAnalyticsWorkspaceName }${uniqueData}', 0, 16)
-var privateLinkScopeName = substring('plscope${logAnalyticsWorkspaceName }${uniqueData}', 0, 16)
-var privateLinkScopeResourceName = substring('plscres${logAnalyticsWorkspaceName }${uniqueData}', 0, 16)
+var privateLinkConnectionName  = take('plconn${logAnalyticsWorkspaceName }${uniqueData}', 80)
+var privateLinkEndpointName = take('pl${logAnalyticsWorkspaceName }${uniqueData}', 80)
+var privateLinkScopeName = take('plscope${logAnalyticsWorkspaceName }${uniqueData}', 80)
+var privateLinkScopeResourceName = take('plscres${logAnalyticsWorkspaceName }${uniqueData}', 80)
```

Comment:
```suggestion
var privateLinkConnectionName  = take('plconn${logAnalyticsWorkspaceName}${uniqueData}', 80)
var privateLinkEndpointName = take('pl${logAnalyticsWorkspaceName}${uniqueData}', 80)
var privateLinkScopeName = take('plscope${logAnalyticsWorkspaceName}${uniqueData}', 80)
var privateLinkScopeResourceName = take('plscres${logAnalyticsWorkspaceName}${uniqueData}', 80)
```

Unless the whitespace is desired? It would be outside of the interpolation designator `${foo} bar`?

Reply:
Since we're touching this file, some [Boy Scoutting](https://www.informit.com/articles/article.aspx?p=1235624&seqNum=6), the `privateDnsZoneId` assignment on L96 has some extra whitespace.

Not a syntax error today, but I could see a future where it is one.

https://github.com/Azure/missionlz/blob/f7d6aa353ea0b814060c5fa19b274e9123fe24fc/src/bicep/modules/privateLink.bicep#L96

Reply:
Since the space in in between the `{}` it is not reflected in the output. For example, the test run ended with: 
![image](https://user-images.githubusercontent.com/8386143/142422128-c18df620-248d-4e54-b171-7f749bdd048f.png)
Notice, no space between "...laws" and "85.." 

Its interesting the second space doesn't cause a problem either in linting (it lets you dot notate with it) or running, that is some extra logic in Bicep I guess that makes it smarter than me :)

Changing any how, since both could be future issues in newer versions.